### ISSUE TITLE
Calling `get_flag` in dependencies results in a fatal error and exit

### ISSUE THREAD

**Issue by gustaff-weldon** [devspace-sh/devspace#2461]
**What happened?**

I have a project that has a hierarchical structure.
Devspace throws a fatal error when using `get_flag` function in a dependency, eg.:

```
❯ cd main-project
❯ devspace deploy

info Using namespace 'devspace'
info Using kube context 'docker-desktop'
sub-project couldn't find flag profile
fatal exit status 1
(main project depends on sub-project)
```

This works fine if I deploy the sub-project directly:

```
❯ cd sub-project
❯ devspace deploy

info Using namespace 'devspace'
info Using kube context 'docker-desktop'
Ensuring image pull secret for registry: hub.docker.com...
build:sub-project Rebuild image sub-project because tag is missing
build:sub-project Building image 'sub-project:AqMuzV' with engine 'buildkit'
build:sub-project Execute BuildKit command with: docker buildx build --tag sub-project:AqMuzV --file Dockerfile -
build:sub-project Sending build context to Docker daemon  6.656kB
... snip docker output ...
build:sub-project Done processing image 'sub-project'
deploy:sub-project Deploying chart /Users/bernard/.devspace/component-chart/component-chart-0.8.5.tgz (sub-project) with helm...
deploy:sub-project Deployed helm chart (Release revision: 1)
deploy:sub-project Successfully deployed sub-project with helm
```

The relevant sections on both projects look like this:

```
pipelines:
  deploy:
    run: |-
      # this fails or not depending from where deploy is called
      profiles=$(get_flag "profile")

      run_dependencies --all
      build_images --all -t ${DEVSPACE_RANDOM}
      create_deployments --all
```

In the example I have used profile, which is global, but it also fails if I try to read deploy-specific flag like build-sequential eg:

```
❯ devspace deploy
info Using namespace 'devspace'
info Using kube context 'docker-desktop'
sub-project couldn't find flag build-sequential
fatal exit status 1
```

**What did you expect to happen instead?**
I expected deployment to work and if flag was set, I would like to see its value in dependency pipeline

**How can we reproduce the bug?** (as minimally and precisely as possible)
repro here: [https://github.com/gustaff-weldon/devspace-dependency-flags](https://github.com/gustaff-weldon/devspace-dependency-flags)

**Local Environment:**

* DevSpace Version: 6.2.1
* Operating System: mac
* ARCH of the OS: AMD64
  **Kubernetes Cluster:**
* Local: Docker Desktop
* Kubernetes Version: v1.25.2

**Anything else we need to know?**

**Comment by gustaff-weldon** [comment]
@pratikjagrut is there any progress on this? This issue causes devspace to fail when deploying with dependencies...

**Comment by gustaff-weldon** [comment]
@lizardruss I'm still getting this error in 6.2.5 using my reproduction repository (linked in description): https://github.com/gustaff-weldon/devspace-dependency-flags

**Comment by gustaff-weldon** [comment]
Ok, looks like the fix has not been released in 6.2.5 https://github.com/devspace-sh/devspace/compare/v6.2.0...v6.2.5
